### PR TITLE
【Hackathon 5 No.16】为 Paddle 新增 EmbeddingBag API

### PR DESCRIPTION
**Pull request by lisamhy**
https://github.com/PaddlePaddle/Paddle/issues/57262

### CODE REVIEW

**Comment by lisamhy** [comment]
@Charles-hit 麻烦review下，谢谢。

File: rfcs/APIs/20231008_api_design_for_embeddingbag.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+with mode="mean" is equivalent to Embedding followed by torch.mean(dim=1),
+
+with mode="max" is equivalent to Embedding followed by torch.max(dim=1).
+```
+
+
+# 四、对比分析
+
+可以直接参考的实现是pytorch，涉及到的API在Paddle中均有实现，可以想到用Paddle API组合实现相同的逻辑
```
Comment (Charles-hit) [inline rfcs/APIs/20231008_api_design_for_embeddingbag.md:57]: 这个任务需要在之前未开发完的PR上接着进行开发，需要手写embeddingbag算子。

File: rfcs/APIs/20231008_api_design_for_embeddingbag.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+
+# 五、方案设计
+
+## 命名与参数设计
+
+ API设计为`paddle.nn.functional.embedding_bag(num_embeddings, embedding_dim, padding_idx=None, sparse=False, weight_attr=None, name=None, mode='mean')`和`paddle.nn.EmbeddingBag(num_embeddings, embedding_dim, padding_idx=None, sparse=False, weight_attr=None, name=None, mode='mean')`
```
Comment (Charles-hit) [inline rfcs/APIs/20231008_api_design_for_embeddingbag.md:64]: 这儿forward函数签名需要也给出来

File: rfcs/APIs/20231008_api_design_for_embeddingbag.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            padding_idx=self._padding_idx,
+            sparse=self._sparse,
+            name=self._name,
+        )
+        if self._mode == "sum":
+            return paddle.sum(out, axis=1)
+        elif self._mode == "mean":
+            return paddle.mean(out, axis=1)
+        elif self._mode == "max":
+            return paddle.max(out, axis=-1)
```
Comment (Charles-hit) [inline rfcs/APIs/20231008_api_design_for_embeddingbag.md:162]: 这儿axis是否应该是1

File: rfcs/APIs/20231008_api_design_for_embeddingbag.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        else:
+            raise ValueError("Not supported mode")
+ ~~~
+
+ # 六、测试和验收的考量
+
+ 测试考虑的case如下：
```
Comment (Charles-hit) [inline rfcs/APIs/20231008_api_design_for_embeddingbag.md:169]: 测试需要对sparse也进行测试，可以写一下完成的测试方案，比如考虑哪些边界条件。

File: rfcs/APIs/20231008_api_design_for_embeddingbag.md (hunk 1 above)
Comment (lisamhy) [reply rfcs/APIs/20231008_api_design_for_embeddingbag.md:57]: 汗。`可以在之前开发者https://github.com/PaddlePaddle/Paddle/pull/49000基础上进行开发。` 这里写的是可以呀！

**Comment by lisamhy** [comment]
@luotao1 @Charles-hit 麻烦确认下是否能组和实现？

File: rfcs/APIs/20231008_api_design_for_embeddingbag.md (hunk 1 above)
Comment (Charles-hit) [reply rfcs/APIs/20231008_api_design_for_embeddingbag.md:57]: 出于性能的考量，这个算子要求写kernel的哈，因为之前已经开发过一版了，需要在那个pr基础上完成后续工作的开发。

**Comment by luotao1** [comment]
@lisamhy 如 https://github.com/PaddlePaddle/community/pull/687#discussion_r1360036691 所说，不能组合实现，需要新增 Kernel
